src/main/java/green/citibike/CitiController.java

```diff
@@ -87,17 +85,19 @@ private double haversine(double lat1, double lon1, double lat2, double lon2) {
     }
 
     private double haversine(String station1, String station2) {
-        double lat1 = stationStatusMap.get(station1).lat;
-        double lat2 = stationStatusMap.get(station2).lat;
-        double lon1 = stationStatusMap.get(station1).lon;
-        double lon2 = stationStatusMap.get(station2).lon;
+        double lat1 = stationStatusMap.get(station1).getLat();
+        double lat2 = stationStatusMap.get(station2).getLat();
+        double lon1 = stationStatusMap.get(station1).getLon();
+        double lon2 = stationStatusMap.get(station2).getLon();
         return haversine(lat1, lon1, lat2, lon2);
     }
 
     private StationStatus findClosestStationId(double lat, double lon, boolean hasBike) {
         String stationId = findClosestStationCoords(lat, lon);
 
-        if (stationStatusMap.get(stationId).num_bikes_available == 0) {
+        int available = hasBike ? stationStatusMap.get(stationId).getNumDocksAvailable()
+                : stationStatusMap.get(stationId).getNumBikesAvailable();
+        if (available == 0) {
             String closestBefore = findClosestStationDirected(stationId, hasBike, true);
             String closestAfter = findClosestStationDirected(stationId, hasBike, false);
```

Comment:
What does this method do?

Reply:
In case the station before the current station in the json is closer than the station afterwards in the case that current station doesn't have bikes it finds the actual closest one unless I'm totally confused

Reply:
If a station doesn't have bikes, then skip over it when looking for the closest station